packages/plugins/src/cli/deploy.ts

```diff
@@ -423,17 +423,32 @@ export const deploy = async (
     }
   }
 
-  const deploymentArtifacts = await makeDeploymentArtifacts(
+  // Create a deployment artifact object that contains the previous contract deployment artifacts,
+  // which will potentially be modified when we make the deployment artifacts. Specifically, the
+  // `history` field of the contract deployment artifacts could be modified. We don't include the
+  // existing execution artifacts or the compiler inputs in this object because we don't need to
+  // modify them.
+  const deploymentArtifacts = {
+    networks: {
+      [chainId.toString()]: {
+        contractDeploymentArtifacts: previousContractArtifacts,
+        executionArtifacts: {},
```

Comment:
Nit: Not including the `executionArtifacts` here could potentially be a footgun if we later add logic that does modify them and forget that we didn't pass them in here. I would probably include them here unless it requires a huge amount of work to do.